py_zipkin/_encoding_helpers.py

```diff
@@ -342,24 +380,32 @@ def fits(self, current_count, current_size, max_size, new_span):
         """
         return 2 + current_count + current_size + len(new_span) <= max_size
 
-    def _create_v1_endpoint(self, endpoint):
-        """Converts an Endpoint to a v1 endpoint dict.
+    def _create_json_endpoint(self, endpoint):
+        """Converts an Endpoint to a JSON endpoint dict.
 
         :param endpoint: endpoint object to convert.
         :type endpoint: Endpoint
-        :return: dict representing a V1 endpoint.
+        :return: dict representing a JSON endpoint.
         :rtype: dict
         """
-        v1_endpoint = {
+        json_endpoint = {
             'serviceName': endpoint.service_name,
```

Comment:
there's a slight difference in v1 vs v2 here. In v1, serviceName could not be null (rather you'd have to hack with empty string).

v2 any value can be left out, though it is poor form to serialize an endpoint with no value.

for this reason of subtlety, I don't tend to share code between the two as documenting why is weirder than the duplicate code :P

Reply:
Even in V1 JSON I should use `""`?

> though it is poor form to serialize an endpoint with no value
Is there really such a big advantage in letting users emit partial or incomplete spans? It seems to me it complicates the model with little benefits. 

At least here at Yelp partial spans are not allowed. They'd mess up our kafka consumers and break our splunk dashboards since those wouldn't be able to merge a partial span to the rest.